### ISSUE TITLE
Python Function: UV can't build with SST dependency due to Git failure

### ISSUE THREAD

**Issue by madisonbullard** [anomalyco/sst#5088]
I'm having trouble executing a Python lambda in sst dev. I'm getting an error from uv when I try to call the function: 
```
|  Build       packages/image-analysis/src/function.handler
|  +2562047h     × Invalid `--with` requirement
|  +2562047h     ╰─▶ Git operation failed
```

The error stems from trying to install SST as a dependency via `pyproject.toml`. If I remove the dependency, it builds without error.

I'm following the aws-python example from this repo, and including this in my `pyproject.toml`:
```toml
[tool.uv.sources]
sst = { git = "https://github.com/sst/sst.git", subdirectory = "sdk/python", branch = "dev" }
```

Seems related to [this](https://github.com/sst/sst/blob/bad21bf409eba1ed308aa2b89e0a8bb473b95e78/pkg/runtime/python/python.go#L321) in the python.go file, maybe?

**Comment by jayair** [comment]
Hmm AFAIK we need that for referencing links in your functions.

**Comment by theoneamendez** [comment]
+1 on fixing this issue. Would love for this to work, in the meantime I just copied and pasted https://github.com/sst/sst/blob/dev/sdk/python/src/sst/__init__.py into src/sst.py and then import as from sst import Resource to unblock myself since the code is not that long.

**Comment by atwmarshall** [comment]
+1 on this issue - would love to see this resolved as caused quite a lot of pain and almost pushed our tech stack away from sst so may prevent other python users from adopting sst.

**Comment by jayair** [comment]
This is an old issue, we've redone our python support since then. Open a new issue if you are still running into this?
